File: app/lib/model/note.dart
class Note {
  final String? id;
  final String title;
  final String body;
  String? created = DateTime.now().toString();
  final String creator;

  Note({this.id, required this.title, required this.body, this.created, required this.creator});

  factory Note.fromJson(Map<String, dynamic> json) {
    return Note(
      id: json['_id'],
      title: json['title'],
      body: json['body'],
      created: json['created'],
      creator: json['creator']
    );
  }

  @override
  String toString() {
    return 'Note{id: $id, title: $title, body: $body, created: $created, creator: $creator}';
  }

  Map toMap() {
    var map = {};
    if (id != null) {
      map["_id"] = id;
    }
    map["title"] = title;
    map["body"] = body;
    map["created"] = created ?? DateTime.now().toString();
    map["creator"] = creator;
    return map;
  }
}

File: schema/Note.js
const mongoose = require('mongoose');

const noteSchema = new mongoose.Schema({
    title: {
        type: String, required: true
    }, body: String,
    created: {
        type: Date, default: Date.now
    }, creator: {
        type: mongoose.Schema.Types.ObjectId, ref: 'creator', required: true
    }, starred: {
        type: Boolean, default: false
    }, upvoters: [mongoose.Schema.Types.ObjectId],
    downvoters: [mongoose.Schema.Types.ObjectId],
    comments: [mongoose.Schema.Types.ObjectId]
})

module.exports = mongoose.model('Note', noteSchema);
